DAOS-7174 pool: connect by label support

kccain opened the pull request:
With this change, libdaos clients may connect to a pool specifying a
pool label. This is an alternative to connecting by UUID. A new API
daos_pool_connect_bylabel() is provided that uses a new CaRT interface
MGMT_POOL_FIND (by label option). Also a new dRPC (engine to server
upcall) SRV_POOL_FIND_BYLABEL is added to fulfill lookups by label.

MGMT_POOL_FIND is universal (supports lookup by UUID or label). Lookup
by UUID calls the existing engine/server dRPC SRV_GET_POOL_SVC.
MGMT_POOL_FIND is meant to replace existing MGMT_POOL_GET_SVCRANKS_RPC
that provides lookup UUID only. The client code for GET_SVCRANKS
is removed with this change, and server code is temporarily kept
in case older libdaos clients connect with the newer engine code.

Also with this change, the daos_test pool properties test creates a
pool with a label property, and exercises daos_pool_connect_bylabel().

Signed-off-by: Kenneth Cain <kenneth.c.cain@intel.com>

daosbuild1 commented on src/engine/drpc_client.c:
```diff
@@ -328,6 +328,87 @@ ds_get_pool_svc_ranks(uuid_t pool_uuid, d_rank_list_t **svc_ranks)
+{
+	struct drpc_alloc		 alloc = PROTO_ALLOCATOR_INIT(alloc);
+	/* FIXME - 82 characters below - sigh */
+	Srv__PoolFindByLabelReq		 frq = SRV__POOL_FIND_BY_LABEL_REQ__INIT;
```
(style)  line over 80 characters

daosbuild1 commented on src/mgmt/rpc.h:
```diff
@@ -124,6 +127,17 @@ CRT_RPC_DECLARE(mgmt_profile, DAOS_ISEQ_MGMT_PROFILE,
+#define DAOS_ISEQ_MGMT_POOL_FIND_BYLABEL /* input fields */	 \
+	((d_const_string_t)	(fbli_label)		CRT_VAR)
+
+#define DAOS_OSEQ_MGMT_POOL_FIND_BYLABEL /* output fields */	 \
```
(style)  Macros with complex values should be enclosed in parentheses

daosbuild1 replied:
(style)  Macros with complex values should be enclosed in parentheses

daosbuild1 replied:
(style)  Macros with complex values should be enclosed in parentheses

daosbuild1 replied:
(style)  Macros with complex values should be enclosed in parentheses

daosbuild1 replied:
(style)  Macros with complex values should be enclosed in parentheses

daosbuild1 commented on src/tests/suite/daos_pool.c:
```diff
@@ -474,11 +474,14 @@ pool_properties(void **state)
-#if 0
-	prop = daos_prop_alloc(2);
+	prop = daos_prop_alloc(1);
+	/* pool label - set arg->pool_label to use daos_pool_connect_bylabel() */
```
(style)  line over 80 characters

daosbuild1 replied:
(style)  line over 80 characters

daosbuild1 reviewed (dismissed):
Style warning(s) for job https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-5635/1/
Please review https://wiki.hpdd.intel.com/display/DC/Coding+Rules

daosbuild1 commented:
Test stage checkpatch completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-5635/1/execution/node/104/log

daosbuild1 reviewed (changes requested):
Style warning(s) for job https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-5635/2/
Please review https://wiki.hpdd.intel.com/display/DC/Coding+Rules

daosbuild1 commented:
Test stage checkpatch completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-5635/2/execution/node/124/log

kccain commented on src/include/daos_task.h:
```diff
@@ -173,12 +173,14 @@ typedef struct {
 typedef struct {
-	/** UUID of the pool. */
-	uuid_t			uuid;
+	/** Label of the pool. May be NULL if connecting by UUID */
+	const char		*label;
+	/** UUID of the pool. May be the NULL UUID if connecting  by label */
+	uuid_t			 uuid;
```
In case it is necessary, I have made a change in my local repo to have a daos_pool_connect_bylabel_t type, and refactored the dc_pool_connect() and (new) dc_pool_connect_bylabel() so that they can still use a lot of common code without copy/paste (allocate / init struct dc_pool and perform the rest of the pool connect procedure).

kccain commented on src/pool/cli.c:
```diff
@@ -217,6 +217,57 @@ dc_pool_choose_svc_rank(const uuid_t puuid, struct rsvc_client *cli,
+ * DER_NOTREPLICA, (clients only) try to refresh the list by querying the MS.
+ */
+int
+dc_pool_choose_svc_rank_bylabel(const char *label, uuid_t puuid,
```
I can consider this next, though I may be tempted to reduce duplication similarly in dc_mgmt_get_pool_svc_ranks() / dc_mgmt_pool_find_bylabel() and ultimately across the CaRT RPCs GET_SVCRANKS and FIND_BYLABEL. And maybe into the dRPC upcalls on the engine/server side. I think all of this would (if some unforeseen complexities are not seen along the way) result in a single "find pool by label or UUID" approach. At which point we might decide to abandon the existing GET_SVCRANKS interface that is by UUID only (or, perhaps leave it on the engine side in case older clients interact with newer servers?).

liw replied:
Yeah, I thought about the similar dup between dc_mgmt_get_pool_svc_ranks and dc_mgmt_pool_find_bylabel. If they use different request and reply types, combining them might be awkward. Don't try _too_ hard, I think.

daosbuild1 commented on src/include/daos_task.h:
```diff
@@ -171,20 +172,34 @@ typedef struct {
 } daos_pool_connect_t;
 
+/** pool connect by label args */
+typedef struct {
```
(style)  do not add new typedefs

daosbuild1 replied:
(style)  do not add new typedefs

daosbuild1 replied:
(style)  do not add new typedefs

daosbuild1 replied:
(style)  do not add new typedefs

daosbuild1 replied:
(style)  do not add new typedefs

daosbuild1 replied:
(style)  do not add new typedefs

daosbuild1 replied:
(style)  do not add new typedefs

daosbuild1 replied:
(style)  do not add new typedefs

daosbuild1 commented on src/pool/cli.c:
```diff
@@ -563,6 +635,86 @@ dc_pool_connect(tse_task_t *task)
+
+	rc = dc_pool_connect_internal(task, args->info, args->poh);
+	if (rc)
+		goto out_pool;;
```
(style)  Statements terminations use 1 semicolon

daosbuild1 commented on src/pool/cli.c:
```diff
@@ -563,6 +635,86 @@ dc_pool_connect(tse_task_t *task)
+
+	rc = dc_pool_connect_internal(task, args->info, args->poh);
+	if (rc)
+		goto out_pool;;
```
(style)  Statements terminations use 1 semicolon

daosbuild1 reviewed (changes requested):
Style warning(s) for job https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-5635/3/
Please review https://wiki.hpdd.intel.com/display/DC/Coding+Rules

daosbuild1 commented:
Test stage checkpatch completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-5635/3/execution/node/99/log

daosbuild1 commented:
Test stage checkpatch completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-5635/4/execution/node/99/log

daosbuild1 commented on src/pool/cli.c:
```diff
@@ -563,6 +635,86 @@ dc_pool_connect(tse_task_t *task)
+
+	rc = dc_pool_connect_internal(task, args->info, args->poh);
+	if (rc)
+		goto out_pool;;
```
(style)  Statements terminations use 1 semicolon

daosbuild1 reviewed (dismissed):
Style warning(s) for job https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-5635/5/
Please review https://wiki.hpdd.intel.com/display/DC/Coding+Rules

daosbuild1 commented:
Test stage checkpatch completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-5635/5/execution/node/100/log

daosbuild1 reviewed (dismissed):
Style warning(s) for job https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-5635/6/
Please review https://wiki.hpdd.intel.com/display/DC/Coding+Rules

daosbuild1 commented:
Test stage checkpatch completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-5635/6/execution/node/100/log

daosbuild1 commented:
Test stage Functional on Leap 15 completed with status FAILURE.  https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-5635/6/display/redirect

daosbuild1 commented on src/mgmt/rpc.h:
```diff
@@ -124,7 +127,22 @@ CRT_RPC_DECLARE(mgmt_profile, DAOS_ISEQ_MGMT_PROFILE,
 		DAOS_OSEQ_MGMT_POOL_GET_SVCRANKS)
 
-#define DAOS_ISEQ_MGMT_TGT_CREATE /* input fields */		 \
+#define DAOS_ISEQ_MGMT_POOL_FIND /* input fields */		 \
```
(style)  Macros with complex values should be enclosed in parentheses

daosbuild1 commented on src/mgmt/rpc.h:
```diff
@@ -124,7 +127,22 @@ CRT_RPC_DECLARE(mgmt_profile, DAOS_ISEQ_MGMT_PROFILE,
+
+#define MGMT_POOL_FIND_DUMMY_LABEL "NO LABEL, FINDING BY UUID"
+
+		#define DAOS_ISEQ_MGMT_TGT_CREATE /* input fields */		 \
```
(style)  line over 80 characters
(style)  Macros with complex values should be enclosed in parentheses

daosbuild1 reviewed (dismissed):
Style warning(s) for job https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-5635/7/
Please review https://wiki.hpdd.intel.com/display/DC/Coding+Rules

daosbuild1 commented:
Test stage checkpatch completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-5635/7/execution/node/85/log

daosbuild1 commented on src/mgmt/rpc.h:
```diff
@@ -124,6 +127,21 @@ CRT_RPC_DECLARE(mgmt_profile, DAOS_ISEQ_MGMT_PROFILE,
 CRT_RPC_DECLARE(mgmt_pool_get_svcranks, DAOS_ISEQ_MGMT_POOL_GET_SVCRANKS,
 		DAOS_OSEQ_MGMT_POOL_GET_SVCRANKS)
 
+#define DAOS_ISEQ_MGMT_POOL_FIND /* input fields */		 \
```
(style)  Macros with complex values should be enclosed in parentheses

daosbuild1 replied:
(style)  Macros with complex values should be enclosed in parentheses

daosbuild1 replied:
(style)  Macros with complex values should be enclosed in parentheses

daosbuild1 replied:
(style)  Macros with complex values should be enclosed in parentheses

daosbuild1 replied:
(style)  Macros with complex values should be enclosed in parentheses

daosbuild1 replied:
(style)  Macros with complex values should be enclosed in parentheses

daosbuild1 commented on src/mgmt/rpc.h:
```diff
@@ -124,6 +127,21 @@ CRT_RPC_DECLARE(mgmt_profile, DAOS_ISEQ_MGMT_PROFILE,
+	((d_const_string_t)	(pfi_label)		CRT_VAR) \
+	((uuid_t)		(pfi_puuid)		CRT_VAR)
+
+#define DAOS_OSEQ_MGMT_POOL_FIND /* output fields */		 \
```
(style)  Macros with complex values should be enclosed in parentheses

daosbuild1 replied:
(style)  Macros with complex values should be enclosed in parentheses

daosbuild1 replied:
(style)  Macros with complex values should be enclosed in parentheses

daosbuild1 replied:
(style)  Macros with complex values should be enclosed in parentheses

daosbuild1 replied:
(style)  Macros with complex values should be enclosed in parentheses

daosbuild1 replied:
(style)  Macros with complex values should be enclosed in parentheses

daosbuild1 reviewed (dismissed):
Style warning(s) for job https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-5635/8/
Please review https://wiki.hpdd.intel.com/display/DC/Coding+Rules

daosbuild1 commented:
Test stage Functional on CentOS 8 completed with status FAILURE.  https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-5635/9/display/redirect

daosbuild1 commented:
Test stage Functional on CentOS 7 completed with status FAILURE.  https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-5635/9/display/redirect

daosbuild1 reviewed (dismissed):
Style warning(s) for job https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-5635/10/
Please review https://wiki.hpdd.intel.com/display/DC/Coding+Rules

daosbuild1 commented:
Test stage checkpatch completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-5635/10/execution/node/98/log

daosbuild1 commented:
Test stage Unit Test completed with status FAILURE.  https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-5635/10/display/redirect

daosbuild1 commented:
Test stage Unit Test with memcheck completed with status FAILURE.  https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-5635/10/display/redirect

daosbuild1 reviewed (dismissed):
Style warning(s) for job https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-5635/11/
Please review https://wiki.hpdd.intel.com/display/DC/Coding+Rules

daosbuild1 commented:
Test stage checkpatch completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-5635/11/execution/node/85/log

daosbuild1 commented:
Test stage Unit Test completed with status FAILURE.  https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-5635/11/display/redirect

daosbuild1 reviewed (dismissed):
Style warning(s) for job https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-5635/12/
Please review https://wiki.hpdd.intel.com/display/DC/Coding+Rules

daosbuild1 commented:
Test stage checkpatch completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-5635/12/execution/node/85/log

daosbuild1 commented on src/include/daos_task.h:
```diff
@@ -171,20 +174,34 @@ typedef struct {
 } daos_pool_connect_t;
 
+/** pool connect by label args */
+typedef struct {
```
(style)  do not add new typedefs

daosbuild1 replied:
(style)  do not add new typedefs

daosbuild1 replied:
(style)  do not add new typedefs

daosbuild1 replied:
(style)  do not add new typedefs

daosbuild1 reviewed (dismissed):
Style warning(s) for job https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-5635/13/
Please review https://wiki.hpdd.intel.com/display/DC/Coding+Rules

daosbuild1 commented:
Test stage checkpatch completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-5635/13/execution/node/98/log

daosbuild1 reviewed (dismissed):
Style warning(s) for job https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-5635/15/
Please review https://wiki.hpdd.intel.com/display/DC/Coding+Rules

daosbuild1 commented:
Test stage checkpatch completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-5635/15/execution/node/98/log

mjmac commented on src/include/daos_task.h:
```diff
@@ -171,20 +174,34 @@ typedef struct {
+	daos_handle_t		*poh;
+	/** Optional, returned pool information. */
+	daos_pool_info_t	*info;
+} daos_pool_connect_lbl_t;
```
Why can't we just extend the existing `daos_pool_connect_t` struct to include the label field?

kccain replied:
Some clients such as the HDF5 daos vol connector use the task API and extending the existing structure causes an ABI break. Changing existing API/ABI is trying to be avoided in this patch. In earlier edits of this patch I had some related problems in this part of the code that resulted in segmentation faults in vol-daos tests.

mchaarawi replied:
technically we could just land this PR with the vol tests disabled, and as soon as the master build becomes available, we can trigger the HDF5 VOL builds so that future PRs won't fail.
(this would also allow to insert the new symbols appropriately in the task table and not just at the end)

kccain replied:
@mchaarawi what would be the implications of adding such an ABI break for daos packaging, and the consumer packages? Would it require new package versions to be built, along with bumping the API version to a new major version 2?

mchaarawi replied:
yes indeed. would require in this case any middleware using the task API to be rebuilt.

kccain replied:
OK, trying to avoid that in this first increment, so propose to keep the patch as currently constructed. Subsequently it is planned to evaluate how tools/middlewares could take advantage of the new API (and, if we do decide to rearrange the task table items, to rebuild for a new ABI).

mchaarawi replied:
my thought was more to remove this new:
daos_pool_connect_lbl_t
and just add an optional 
const char *label
to
daos_pool_connect_t

I do understand the issue with the HDF5 vol. does it still break if you add the label as the last element in the struct?

mchaarawi replied:
fwiw im fine to land this as is now, and do such changes in a future PR. would just need to avoid releasing 2.0 before making that change if we want it.

kccain replied:
I'm trying the simplification to put const char *label at the end of daos_pool_connect_t in an experimental PR that will run just the HDF5 vol tests and daos_core_test pool tests. https://github.com/daos-stack/daos/pull/5748

kccain replied:
That experiment seemed to work (daos_vol tests passed). I propose let's land this PR in current form, and I will try to roll this task API simplification into the follow-up PR for this work, https://github.com/daos-stack/daos/pull/5727 . That PR also adds container open bylabel API (and while doing that work I can also verify the simplification applies to dc_cont_open_t and won't require a new dc_cont_open_lbl_t).

mjmac reviewed (commented):
Took a quick look through, looks pretty good. The only big question I have is why we need a new connection type. I'm probably missing something, but it seems like we could just add the label field to the existing type and then switch based on whether the uuid or label was supplied.

kccain commented on src/include/daos/pool.h:
```diff
@@ -102,8 +104,12 @@ int dc_pool_list_cont(tse_task_t *task);
 int dc_pool_map_version_get(daos_handle_t ph, unsigned int *map_ver);
 int dc_pool_update_map(daos_handle_t ph, struct pool_map *map);
-int dc_pool_choose_svc_rank(const uuid_t puuid, struct rsvc_client *cli,
+int dc_pool_choose_svc_rank_ORIG(const uuid_t puuid, struct rsvc_client *cli,
```
Yes, will remove in next push, thanks

liw commented on src/mgmt/rpc.h:
```diff
@@ -124,6 +127,21 @@ CRT_RPC_DECLARE(mgmt_profile, DAOS_ISEQ_MGMT_PROFILE,
 		DAOS_OSEQ_MGMT_POOL_GET_SVCRANKS)
 
+#define DAOS_ISEQ_MGMT_POOL_FIND /* input fields */		 \
+	((int32_t)		(pfi_bylabel)		CRT_VAR) \
+	((d_const_string_t)	(pfi_label)		CRT_VAR) \
```
Do we need a 32-bit padding in between?

kccain replied:
Yes I think so, will add

kccain commented on src/include/daos_pool.h:
```diff
@@ -190,6 +190,34 @@ daos_pool_connect(const uuid_t uuid, const char *grp,
 		  daos_handle_t *poh, daos_pool_info_t *info, daos_event_t *ev);
 
+/**
+ * Connect to the DAOS pool identified by \a label property. Upon a successful
```
Will remove the word property in this documentation. Probably want to keep the label property and yes let's consider (in a future patch before or when we implement rename) disabling the ability to set that property. The decision on label property has impact on the create API that accepts properties input and is used in the current implementation to know if a pool has a user-specified name. If we removed label as a property altogether, then would need a new create function I would think (vs. a very disruptive alternative of a major API/ABI change to the existing pool create function).

kccain commented on src/client/api/pool.c:
```diff
@@ -38,6 +38,36 @@ daos_pool_connect(const uuid_t uuid, const char *grp,
+	if (label)
+		label_len = strnlen(label, DAOS_PROP_LABEL_MAX_LEN+1);
+	if (!label || (label_len == 0) || (label_len > DAOS_PROP_LABEL_MAX_LEN))
+		return -DER_INVAL;
```
Yes, will add it in next push

kccain commented on src/control/server/mgmt_drpc.go:
```diff
@@ -109,6 +111,31 @@ func (mod *srvModule) handleGetPoolServiceRanks(reqb []byte) ([]byte, error) {
+		resp.Status = int32(drpc.DaosNonexistant)
+		mod.log.Debugf("PoolFindByLabelResp: %+v", resp)
+		return proto.Marshal(resp)
+		// return nil, err
```
Will remove in next push, along with another instance found just above in handleGetPoolServiceRanks

kccain commented on src/include/daos/pool.h:
```diff
@@ -53,6 +53,7 @@ struct dc_pool {
 	pthread_rwlock_t	dp_co_list_lock;
 	/* pool uuid */
 	uuid_t			dp_pool;
+	char		       *dp_label;
```
I'm not sure. It may be possible but I think would require unwinding some code refactoring that was done. I think it is probably manageable here, but in a follow-on PR-5727 for container open bylabel, a struct dc_cont field "dc_label" is added too. There the refactoring is a little broader both in the dc_cont_open[_lbl]() functions and the completion callback cont_open_complete().

In this PR:
dc_pool_connect_internal() is used by both dc_pool_connect() and dc_pool_connect_lbl(). The internal function cannot call dc_task_get_args(task) to get an args structure, because there are 2 structure types now (daos_pool_connect_t and daos_pool_connect_lbl_t). So one way I found to refactor the code into this internal function was to store a label in struct dc_pool (gotten via dc_task_set/get_priv). We could decide to copy/paste the internal logic back into dc_pool_connect() and dc_pool_connect_lbl(). 

Other uses seem less of a challenge. e.g., all dc_pool_choose_svc_rank() calls for container and pool operations after a pool has been connected could simply specify a NULL label argument rather than pool->dp_label. i.e., the implementation would always choose a svc rank by providing UUID only.

dc_pool_choose_svc_rank() itself would need a small change too, to test whether the label argument is NULL, in a few places (i.e., avoid debug printing with %s if label is NULL).

liw replied:
We could also add a label parameter to dc_pool_connect_internal, ..., or couldn't we?

liw replied:
If you don't share the feeling about relabel support, then there's no need to make this change.

kccain replied:
I'm looking into adding label parameter to dc_pool_connect_internal() and also whether I can do something similar for the follow-on PR-5727 for container open bylabel (struct dc_cont similar issue).

kccain commented:
Build 15 single test failure appears to be an instance of known bug DAOS-7350.
Will push another change that addresses all comments except the one outstanding question of whether to modify struct dc_pool with a new label field, or not.

daosbuild1 commented on src/mgmt/rpc.h:
```diff
@@ -124,6 +127,22 @@ CRT_RPC_DECLARE(mgmt_profile, DAOS_ISEQ_MGMT_PROFILE,
 CRT_RPC_DECLARE(mgmt_pool_get_svcranks, DAOS_ISEQ_MGMT_POOL_GET_SVCRANKS,
 		DAOS_OSEQ_MGMT_POOL_GET_SVCRANKS)
 
+#define DAOS_ISEQ_MGMT_POOL_FIND /* input fields */		 \
```
(style)  Macros with complex values should be enclosed in parentheses

daosbuild1 replied:
(style)  Macros with complex values should be enclosed in parentheses

daosbuild1 commented on src/mgmt/rpc.h:
```diff
@@ -124,6 +127,22 @@ CRT_RPC_DECLARE(mgmt_profile, DAOS_ISEQ_MGMT_PROFILE,
+	((d_const_string_t)	(pfi_label)		CRT_VAR) \
+	((uuid_t)		(pfi_puuid)		CRT_VAR)
+
+#define DAOS_OSEQ_MGMT_POOL_FIND /* output fields */		 \
```
(style)  Macros with complex values should be enclosed in parentheses

daosbuild1 replied:
(style)  Macros with complex values should be enclosed in parentheses

daosbuild1 reviewed (changes requested):
Style warning(s) for job https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-5635/16/
Please review https://wiki.hpdd.intel.com/display/DC/Coding+Rules

daosbuild1 commented:
Test stage checkpatch completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-5635/16/execution/node/98/log

daosbuild1 reviewed (changes requested):
Style warning(s) for job https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-5635/17/
Please review https://wiki.hpdd.intel.com/display/DC/Coding+Rules

daosbuild1 commented:
Test stage checkpatch completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-5635/17/execution/node/85/log

kccain commented:
This PR's build 17 again affected by daos_test -X DEGRADE23 test case failure https://jira.hpdd.intel.com/browse/DAOS-7350
